source/storage/storage.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"os/exec"
+)
+
+// Source implements FeatureSource.
+type Source struct{}
+
+// Name returns an identifier string for this feature source.
+func (s Source) Name() string { return "storage" }
+
+// Discover returns feature names for storage: nonrotational if any SSD drive present.
+func (s Source) Discover() ([]string, error) {
+	features := []string{}
+
+	// Check is there zero in any device rotational status
+	cmd := exec.Command("/bin/sh", "-c", "grep 0 /sys/block/*/queue/rotational")
```

Comment:
Generally, instead of spawning shell and using grep, it is more safe to write small function to traverse directories and read files.

Reply:
Thanks for suggestion, I rewrote that piece to walk directories instead of grep (previous variant remained visible in kad's comment as "outdated").